src/ServiceClient/Implements/DeribitSocketConnection.cs

```diff
@@ -0,0 +1,63 @@
+﻿using ServiceClient.Abstractions;
+using ServiceClient.DTOs;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ServiceClient.Implements
+{
+    public class DeribitSocketConnection : IDeribitSocketConnection
+    {
+        private readonly WebSocketOptions options;
+
+        private ClientWebSocket? socket;
+
+        public DeribitSocketConnection(WebSocketOptions options) 
+        { 
+            this.options = options;
+        }
+
+        public ClientWebSocket ClientWebSocket
+        {
+            get
+            {
+                if (socket != null)
+                    return socket;
+                socket = new ClientWebSocket
+                {
+                    Options =
+                    {
+                        KeepAliveInterval = TimeSpan.FromSeconds(options.KeepAliveIntervalInSeconds)
+                    }
+                };
+                return socket;
+            }
+        }
+
+        public async Task SocketConnectAsync(CancellationToken cancellationToken)
```

Comment:
Signature should be Task<bool>

Add try catch.. return true if could be connected or false if not.